promoter/promoter.go

```diff
@@ -0,0 +1,92 @@
+package promoter
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"gitlab.com/SkynetLabs/skyd/node/api/client"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type (
+	// Health contains health information about the promoter. Namely the
+	// database and skyd. If everything is ok all fields are 'nil'.
+	// Otherwise the corresponding fields will contain an error.
+	Health struct {
+		Database error
+		Skyd     error
+	}
+
+	// Promoter is a wrapper around a skyd and a database client. It makes
+	// sure that skyd watches all the siacoin addresses it is supposed to
+	// and is capable of adding new addresses to watch and removing old
+	// ones. It can also track the incoming funds that users have sent to
+	// their assigned addresses.
+	Promoter struct {
+		staticClient              *mongo.Client
```

Comment:
nit: It's a bit odd to keep both a client and a DB instance. You can get the client from the DB when you need it.